app/controllers/application_controller.rb

```diff
@@ -1,9 +1,12 @@
 class ApplicationController < ActionController::Base
-  private
+  protect_from_forgery with: :exception
 
-  def current_user
-    @current_user ||= User.first
-  end
+  before_action :update_allowed_parameters, if: :devise_controller?
+
```

Comment:
- [OPTIONAL] Great work configuring `devise` for user authentication. However, I noticed that although the confirmation link is sent to users, users can navigate the site and create posts before confirming their emails. In terms of security, and ensuring a user was the one that signed up, this is not the best implementation. To fix this, I would advise you to use this line of code after the `protect_from_forgery with: :exception` code:

```ruby
before_action :authenticate_user!
``` 

The above code will ensure that users confirm their emails before any action can be carried out. 👍 💻